core/aws_ddk_core/stages/glue_transform.py

```diff
@@ -93,12 +101,13 @@ def __init__(
         # If Glue job name is not supplied, create one
         self._job: Optional[IJob] = None
         if not job_name:
-            self._job = GlueFactory.job(
+            self._job = GlueFactory.job(  # type: ignore
                 self,
                 id=f"{id}-job",
                 environment_id=environment_id,
                 executable=executable,
                 role=job_role,
+                **glue_job_args,
```

Comment:
Looks good. My one concern though is what happens if one of the parameters is redefined in the `glue_job_args`. For example, if `glue_job_args` contains a parameter called `role`, what will happen here exactly?

Will this call to the factory fail because `role` was passed twice? Or will the argument from `glue_job_args` overwrite the one from line 109?

The former scenario would be fine, the latter wound't be good. I wonder if we should make this more explicit, and check for this scenario in the beginning of the function. Like if `glue_job_args` contains `role` or `executable`, maybe we should throw an exception ourselves. The same thing goes for the Glue Crawler args as well.

Reply:
This would raise a Type Error

For example:
```
TypeError: job() got multiple values for keyword argument 'role'

aws_ddk_core/stages/glue_transform.py:104: TypeError
```

Do you think we should pop overwritten values from the dictionary or just let CDK raise the exception?

Reply:
I think this is good, as long as an exception is raised. It might be preferable to raise the exception ourselves, just so we can provide our own error. For example:

```python
if "role" in glue_job_args:
  raise TypeError("Parameter role should not be passed under glue_job_args, but rather as a separate parameter for the stage")
```